app/controllers/admin/notes_controller.rb

```diff
@@ -3,14 +3,14 @@
 module Admin
   class NotesController < ApplicationController
     def create
-      @submission = Submission.find(note_params[:submission_id])
-      note = Note.new(gravity_user_id: @current_user, **note_params)
+      note_attrs = note_params.merge(gravity_user_id: @current_user)
+      note = Note.new(note_attrs)
+      path = admin_submission_path(note.submission)
```

Comment:
is there any case where `note.submission` could fail in a way that `Submission.find(id)` would not have above?

Reply:
If there is, let's get a test in for it!! 😍 But I don't know of one.

Reply:
Are we not loading the parent `submission`? It's important to respect the hierarchy otherwise it could create edge cases we may not have to cover otherwise:

```ruby
before_action :load_submission, only: :create

def create
  note = @submission.notes.new(note_params)
  path = admin_submission_path(@submission)

  if note.save
    ...
  else
    ...
  end
end

def load_submission
  # This could be inlined in #create
  @submission = Submission.find(params.dig(:note, :submission_id))
end

def note_params
  params.require(:note).permit(:submission_id, :body).merge(gravity_user_id: @current_user)
end
```

Reply:
Yeah, that's even better, good call @yuki24! 👍